src/pypdfium2/_helpers/page.py

```diff
@@ -495,6 +521,64 @@ def _auto_bitmap_format(page, fill_color, grayscale, prefer_bgrx, maybe_alpha):
         return pdfium_c.FPDFBitmap_BGR
 
 
+def _check_missing_fonts(page):
+    """
+    Check for non-embedded fonts on the page that may not render correctly.
+    
+    Returns:
+        dict[str, list[str]]: Dictionary mapping font names (base names) to lists of text content that uses them.
+    """
+    missing_fonts = {}
+    textpage = None
+    try:
+        # Create a textpage to extract text content from text objects
+        textpage = page.get_textpage()
+        
+        for obj in page.get_objects(filter=[pdfium_c.FPDF_PAGEOBJ_TEXT]):
+            try:
+                font = obj.get_font()
+                if not font.is_embedded():
+                    font_name = font.get_base_name()
+                    
+                    # Try to extract text content
+                    text_content = None
+                    try:
+                        # Create a PdfTextObj with textpage to extract text
+                        # We create a new wrapper with textpage since the original obj doesn't have one
+                        text_obj_with_textpage = PdfTextObj(obj.raw, textpage=textpage)
```

Comment:
WDYT, should we maybe add a `textpage` passthrough option to `PdfPage.get_objects()` to avoid the need to re-instantiate the `PdfTextObj`s ?